crates/payload/basic/src/lib.rs

```diff
@@ -967,6 +990,49 @@ fn commit_withdrawals<DB: Database<Error = Error>>(
     })
 }
 
+/// TODO: docs
+fn pre_block_contract_call<DB>(
+    db: State<DB>,
```

Comment:
I think this could be `Database + DatabaseCommit`, so it no longer requires a concrete type,
all of them are also implemented on &mut 
so we don't need to return it

Reply:
same in separate good first issue

Reply:
was an easy change, it's all &mut now